wordpress-mobile/WordPress-FluxC-Android
# Woo: create order

This PR adds the ability to create orders from the app, we support currently just: status, a list of products, and customer addresses (shipping and billing).

I created a separate class for the request [CreateOrderRequest](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/218d9523bd33cf04b5e0f151354cfe2e32aa8ab2/plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/order/CreateOrderRequest.kt#L3) because:
1. WCOrderModel has a ton of stuff that we won't need for the creation, and we need to keep passing null arguments for it.
2. WCOrderModel doesn't play well with the OrderAddress type.
3. The new class would make it more flexible in organizing the classes of different sections later on (for example grouping the payment options in the same class).

The new class meant that I had to move the subclasses of WCOrderModel to their own files (for a better organization), sorry @wzieba for this, I know it will create conflicts with your Room changes.

### Testing
1. Open the example app.
2. Navigate to Woo then Orders.
3. Select a site.
4. Click on create order.
5. Enter the order's information.
6. Confirm that the created order's id is printed.
7. Check the order in wp-admin, and confirm that it's matching what you entered.
